wen-restart/src/wen_restart.rs

```diff
@@ -871,6 +872,38 @@ pub(crate) fn aggregate_restart_heaviest_fork(
     }
 }
 
+pub(crate) fn receive_restart_heaviest_fork(
+    wen_restart_coordinator: Pubkey,
+    cluster_info: Arc<ClusterInfo>,
+    exit: Arc<AtomicBool>,
+    progress: &mut WenRestartProgress,
+) -> Result<(Slot, Hash)> {
+    let mut cursor = solana_gossip::crds::Cursor::default();
+    loop {
+        if exit.load(Ordering::Relaxed) {
+            return Err(WenRestartError::Exiting.into());
+        }
+        for new_heaviest_fork in cluster_info.get_restart_heaviest_fork(&mut cursor) {
+            if new_heaviest_fork.from == wen_restart_coordinator {
+                info!(
+                    "Received new heaviest fork from coordinator: {} {:?}",
+                    wen_restart_coordinator, new_heaviest_fork
+                );
+                let coordinator_heaviest_slot = new_heaviest_fork.last_slot;
+                let coordinator_heaviest_hash = new_heaviest_fork.last_slot_hash;
+                progress.coordinator_heaviest_fork = Some(HeaviestForkRecord {
+                    slot: coordinator_heaviest_slot,
+                    bankhash: coordinator_heaviest_hash.to_string(),
+                    total_active_stake: 0,
+                    wallclock: new_heaviest_fork.wallclock,
+                    shred_version: new_heaviest_fork.shred_version as u32,
+                });
+                return Ok((coordinator_heaviest_slot, coordinator_heaviest_hash));
+            }
```

Comment:
maybe should have a sleep here to avoid constantly spinning in this loop

Reply:
Good catch, added.